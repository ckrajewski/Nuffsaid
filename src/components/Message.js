import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';

export default function Message(props) {
	const {message, priority} = props;
	debugger;
	return (
<div> 
<Paper> {message} </Paper>
</div>
		)
}