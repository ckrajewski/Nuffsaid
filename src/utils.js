export const messageThemes = {
  1: '#F56236',
  2: '#FCE788',
  3: '#88FCA3',
};
