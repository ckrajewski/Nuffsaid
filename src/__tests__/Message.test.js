import React from 'react';
import { messageThemes } from '../utils';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Message from '../components/Message';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const priorityThemeTest = (priority) => {
	const message = 'hello';
    const tree = renderer.create(<Message message={message} priority={priority} />).toJSON();
    expect(tree).toMatchSnapshot();
}

describe('Message component', () => {
  test('renders', () => {
    const wrapper = shallow(<Message />);
    expect(wrapper.exists()).toBe(true);
  });
  test('renders correct message', () => {
  	const message = 'hello';
    const tree = renderer
    .create(<Message message={message} priority={1} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders correct error message theme', () => {
  	const priority = 1;
  	priorityThemeTest(priority);
  });
  test('renders correct warning message theme', () => {
  	const priority = 2;
  	priorityThemeTest(priority);
  });
  test('renders correct info message theme', () => {
  	const priority = 3;
  	priorityThemeTest(priority);
  });
 });