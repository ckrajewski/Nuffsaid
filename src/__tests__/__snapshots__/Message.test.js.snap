// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders correct error message theme 1`] = `
<div>
  <div
    className="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation5 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": "#F56236",
      }
    }
  >
    <div
      className="MuiGrid-root makeStyles-grid-6 MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root makeStyles-message-4 MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          hello
        </p>
      </div>
      <div
        className="MuiGrid-root makeStyles-clear-5 MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-2"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message component renders correct info message theme 1`] = `
<div>
  <div
    className="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation5 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": "#88FCA3",
      }
    }
  >
    <div
      className="MuiGrid-root makeStyles-grid-6 MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root makeStyles-message-4 MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          hello
        </p>
      </div>
      <div
        className="MuiGrid-root makeStyles-clear-5 MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-2"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message component renders correct message 1`] = `
<div>
  <div
    className="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation5 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": "#F56236",
      }
    }
  >
    <div
      className="MuiGrid-root makeStyles-grid-6 MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root makeStyles-message-4 MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          hello
        </p>
      </div>
      <div
        className="MuiGrid-root makeStyles-clear-5 MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-2"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message component renders correct warning message theme 1`] = `
<div>
  <div
    className="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation5 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": "#FCE788",
      }
    }
  >
    <div
      className="MuiGrid-root makeStyles-grid-6 MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root makeStyles-message-4 MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          hello
        </p>
      </div>
      <div
        className="MuiGrid-root makeStyles-clear-5 MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-2"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
